c-scape/src/fs/dir/readdir.rs

```diff
@@ -93,9 +93,57 @@ unsafe extern "C" fn readdir64(dir: *mut c_void) -> *mut libc::dirent64 {
 }
 
 #[no_mangle]
-unsafe extern "C" fn readdir() {
-    //libc!(libc::readdir());
-    unimplemented!("readdir")
+unsafe extern "C" fn readdir(dir: *mut libc::DIR) -> *mut libc::dirent {
```

Comment:
Interesting. The presence of undefined references to `readdir` on GLIBC indicates a code which has not been compiled with `-D_FILE_OFFSET_BITS=64` or equivalent. In this case, it looks like the code in question is libgit2, so I've now filed https://github.com/rust-lang/git2-rs/pull/862 to fix this. With that PR, libgit2 will use `readdir64` instead of `readdir`, on GLIBC.

I wonder if it makes sense for mustang to leave `readdir` undefined, when `target_env = "gnu"`, to help catch programs that haven't enabled `_FILE_OFFSET_BITS=64`/`_TIME_BITS=64` yet.